cdb_test.go

```diff
@@ -2,216 +2,156 @@ package cdb
 
 import (
 	"hash/fnv"
+	"io/ioutil"
 	"os"
 	"strconv"
 	"sync"
 	"testing"
+
+	"github.com/stretchr/testify/suite"
 )
 
-func TestShouldReturnAllValues(t *testing.T) {
-	f, _ := os.Create("test.cdb")
-	defer f.Close()
-	defer os.Remove("test.cdb")
+type testCDBRecord struct {
+	key []byte
+	val []byte
+}
+type CDBTestSuite struct {
+	suite.Suite
+	cdbFile     *os.File
+	cdbHandle   *CDB
+	testRecords []testCDBRecord
+}
 
-	handle := New()
+func TestCDBTestSuite(t *testing.T) {
+	suite.Run(t, new(CDBTestSuite))
+}
 
-	cases := []struct {
-		key, value string
-	}{
-		{"key1", "value1"},
-		{"key2", "value2"},
-		{"key3", "value3"},
-		{"key4", "value4"},
-		{"key5", "value5"},
-		{"key6", "value6"},
-		{"key7", "value7"},
+func (suite *CDBTestSuite) getCDBReader() Reader {
+	// initialize reader
+	reader, err := suite.cdbHandle.GetReader(suite.cdbFile)
+	if err != nil {
+		suite.T().Errorf("Can't get CDB reader: %#v", err)
+		suite.T().FailNow()
 	}
+	return reader
+}
 
-	writer, err := handle.GetWriter(f)
+func (suite *CDBTestSuite) getCDBWriter() Writer {
+	// initialize writer
+	writer, err := suite.cdbHandle.GetWriter(suite.cdbFile)
 	if err != nil {
-		t.Error(err)
+		suite.T().Errorf("Can't get CDB writer: %#v", err)
+		suite.T().FailNow()
 	}
+	return writer
+}
 
-	for _, c := range cases {
-		writer.Put([]byte(c.key), []byte(c.value))
+func (suite *CDBTestSuite) SetupTest() {
+	f, err := ioutil.TempFile("", "test_*.cdb")
+	if err != nil {
+		suite.T().Errorf("Can't open temporary file: %#v", err)
+		suite.T().FailNow()
 	}
 
-	writer.Close()
-	reader, _ := handle.GetReader(f)
-
-	for _, c := range cases {
-		value, err := reader.Get([]byte(c.key))
-		if err != nil {
-			t.Error(err)
-		}
-
-		if string(value) != c.value {
-			t.Errorf("Expected value %s, got %s", c.value, value)
-		}
-	}
+	suite.cdbFile = f
+	suite.cdbHandle = New()
 
-	if reader.Size() != 7 {
-		t.Errorf("Expected size %d, got %d", 7, reader.Size())
+	// generate test records
+	const testRecordsCount = 10
```

Comment:
Do we need this `const` here? I guess we can remove it

Reply:
fidex